src/vehicle.cpp

```diff
@@ -6923,6 +6845,53 @@ std::string vehicle_part::name() const {
     return res;
 }
 
+itype_id vehicle_part::ammo_current() const
+{
+    // @todo currently only support fuel tanks and batteries
+    return info().has_flag( VPFLAG_FUEL_TANK ) ? info().fuel_type : "null";
+}
+
+long vehicle_part::ammo_capacity() const
+{
+    // @todo currently only support fuel tanks and batteries
+    return info().has_flag( VPFLAG_FUEL_TANK ) ? info().size : 0;
+}
+
+long vehicle_part::ammo_remaining() const
+{
+    // @todo currently only support fuel tanks and batteries
+    return info().has_flag( VPFLAG_FUEL_TANK ) ? amount : 0;
+}
+
+int vehicle_part::ammo_set( const itype_id &ammo, long qty )
+{
+    // @todo currently only support fuel tanks and batteries
+    if( !info().has_flag( VPFLAG_FUEL_TANK ) ) {
+        return -1;
+    }
+
+    if( ammo == "null" && qty == 0 ) {
+        return amount = 0;
```

Comment:
I don't like constructs like that. Saves one line, but adds one "WTF".
Same at the end of the function.